### PR TITLE
FIX: Prevent negative column width, row height

### PR DESCRIPTION
**Pull request by enjoythecode**
fixes !165

### CODE REVIEW

**Comment by enjoythecode** [comment]
fixes ironcalc/ironcalc!165

**Comment by enjoythecode** [comment]
hmm, I can't reference that issue apparently.

**Comment by enjoythecode** [comment]
fixes #165

there we go.

**Comment by enjoythecode** [comment]
@nhatcher  do you have a preferred way of reaching out to you when a PR is ready for review?

this is ready for your review, thanks for your guidance.

File: base/src/test/test_column_width.rs [hunk 1]
```diff
@@ -82,3 +82,26 @@ fn test_column_width_higher_edge() {
...
+fn test_column_width_negative() {
+    let mut model = new_empty_model();
+    let cols = vec![Col {
+        custom_width: false,
+        max: 16,
+        min: 5,
+        style: Some(1),
+        width: 10.0,
+    }];
+    model.workbook.worksheets[0].cols = cols;
```
Comment (nhatcher) [inline base/src/test/test_column_width.rs]: You don't need to do this, the default column widths will do.

File: base/src/test/test_column_width.rs [hunk 2]
```diff
@@ -82,3 +82,26 @@ fn test_column_width_higher_edge() {
...
+    let result = model
+        .workbook
+        .worksheet_mut(0)
+        .unwrap()
+        .set_column_width(16, -1.0);
+    assert!(result.is_err());
```
Comment (nhatcher) [inline base/src/test/test_column_width.rs]: Could you do something like:

```rust
assert_eq!(result, Err("some error message".to_string());
```

The reason is hat sometimes we assume it is an error but there is actually a different one

File: webapp/src/components/worksheet.tsx [hunk 3]
```diff
@@ -104,10 +104,12 @@ function Worksheet(props: {
         editor: editor,
       },
       onColumnWidthChanges(sheet, column, width) {
+        if (width < 0) return;
```
Comment (nhatcher) [inline webapp/src/components/worksheet.tsx]: Could you use:
```js
if (width < 0) {
  return;
}
```

I will add this rule to biome some day :S

**Review by nhatcher — CHANGES_REQUESTED** [review]
Excellent job.

I let a couple of minor comments

**Comment by enjoythecode** [comment]
Thanks, addressed the comments.

**Comment by nhatcher** [comment]
Build is failing because of a formatting issue (just run `cargo fmt`)

**Review by nhatcher — APPROVED** [review]
LGTM!

**Pull request merged**